### ISSUE TITLE
Passing parameters as query parameters OR as body's properties?

### ISSUE THREAD

**Issue by castarco** [raml-org/raml-spec#468]
Hello,

is there a way to specify that a parameter MUST be present, BUT not enforcing the way we pass it to the API? When I'm talking about "the way to pass the parameter" I'm referring to two possibilities:
  * Passing the parameter as a query string parameter, the typical... `?paramName=paramValue` in the URI.
  * Passing the parameter as a property of a (for example) JSON object inside the request body.

Thank you for your time and wisdom!

**Comment by sichvoge** [comment]
So you'd like to define a required parameter that can be either passed inside a query parameter or body?

**Comment by castarco** [comment]
Yes, that's the idea. This enables the possibility of using/testing the API for simple cases without having to use "special" tools (like curl, or postman), only using a browser... but at the same time allowing an "advanced" mode.
